models/fetchAllEndPointsInfo.model.js

```diff
@@ -0,0 +1,13 @@
+const fs = require('fs/promises')
+
+async function fetchAllEndPointsInfo () {
+   const endpoints = await fs.readFile("./endpoints.json", "utf-8")
+   const parsedEndpoints = JSON.parse(endpoints);
+   if(parsedEndpoints != undefined) {
+   return parsedEndpoints
+   } else {
+   return Promise.reject({ status: 404, msg: "Not Found"})
+   }
+}
+
```

Comment:
I'm not sure you would reach a point of needing to do a Promise.reject here. If the file doesn't exist then readFile would throw an error - but we know that the file does exist. Whenever we have code that returns a rejected promise, or more specifically an error, it should be tested for too. I don't think you can replicate that error really so I don't think it's necessary.